src/commands/announce.ts

```diff
@@ -0,0 +1,43 @@
+import { SlashCommandBuilder } from '@discordjs/builders';
+import { CommandInteraction } from 'discord.js';
+import { PermissionFlagsBits } from 'discord-api-types/v9';
+
+/**
+ * Makes an announcement into a channel.
+ */
+export const data = new SlashCommandBuilder()
+    .setName('announce')
+    .setDescription('Announces a message to a channel.')
+    .setDefaultMemberPermissions(PermissionFlagsBits.Administrator)
+    .addStringOption((option) =>
+        option
+            .setName('message')
+            .setDescription('Your message to be announced.')
+            .setRequired(true)
+    );
+
+export async function execute(interaction: CommandInteraction) {
+    const message = interaction.options.getString('message');
+
+    if (!message) {
+        return interaction.reply({
+            content: 'You need to input a message.',
+            ephemeral: true,
+        });
+    }
+
+    if (message.length > 2000) {
+        return interaction.reply({
+            content: 'Message is too long.',
+            ephemeral: true,
+        });
+    }
+
+    return interaction.reply(message).catch((error: unknown) => {
```

Comment:
This in my opinion has two ways of handling
1) Have reply to the user ephemeral and message is send by using `channel.send()`
```suggestion
    return interaction.channel?.send(message).then(() => interaction.reply({ content: "Message sent", ephemeral: true })).catch((error: unknown) => {
        console.error(error);
        interaction.reply({
            content: 'There was an error trying to announce in this channel!',
            ephemeral: true,
        });
    });
```
2) sending reply and then deleting it
```suggestion
    return interaction.channel?.send(message).then(() => interaction.reply({ content: "Message sent" }).then(() => interaction.deleteReply())).catch((error: unknown) => {
        console.error(error);
        interaction.reply({
            content: 'There was an error trying to announce in this channel!',
            ephemeral: true,
        });
    });```
I personally prefer the first approach since I feel like it is cleaner implementation and you don't have to mess with deleting messages.